src/models/geograph.py

```diff
@@ -43,20 +123,179 @@ def graph(self) -> nx.Graph:
     def graph(self, new_graph):
         self._graph = new_graph
 
-    def _load_graph(self, graph_path: str) -> None:
+    @property
+    def rtree(self) -> STRtree:
+        """Return Rtree object."""
+        return self._rtree
+
+    def _load_vector(
+        self,
+        vector_path: pathlib.Path,
+        attributes: Optional[List[str]] = None,
+        load_slice=None,
+    ) -> STRtree:
+        """
+        Load graph and rtree with vector data from GeoPackage or shape file.
+
+        Args:
+            vector_path (pathlib.Path): Path to a gpkg or shp file.
+            attributes (Optional[List[str]], optional): columns of the dataframe
+            that are added to nodes of graph as attributes. Defaults to None,
+            which will cause all attributes to be added.
+            load_slice: A slice object denoting the rows of the dataframe to
+            load. Defaults to None, meaning load all rows.
+
+        Raises:
+            ValueError: If `vector_path` is not a GPKG or Shapefile.
+
+        Returns:
+            STRtree: The Rtree object used to build the graph.
+        """
+        if vector_path.suffix not in (".shp", ".gpkg"):
+            raise ValueError("Argument `vector_path` should be a GPKG or Shapefile.")
+        # First try to load as GeoPackage, then as Shapefile.
+        if slice is not None:
+            dataframe = gpd.read_file(
+                vector_path, rows=load_slice, enabled_drivers=["GPKG", "ESRI Shapefile"]
```

Comment:
Out of curiosity, what does the `load_slice` do? Is that if we wanted to only load a few subelements of a dataframe?

Reply:
Yes, you can pass a python slice object using the `slice()` constructor https://www.programiz.com/python-programming/methods/built-in/slice